### DISCUSSION TITLE
Email notifications in Kubernetes Deployment

### DISCUSSION THREAD

**Discussion by guptachandresh** [DefectDojo/django-DefectDojo#9204] [General]
Hi All,

I have been trying to deploy DefectDojo using Kubernetes. As I am not from a development background, I had issues initially with it but managed to deploy it along with mailhog as per some earlier discussions that I have seen. But still, I am unable to send out email notifications. I checked the logs & it gives me an error of Connection Refused. A simple python script confirms there is no issue with connectivity & emails can be sent out.

Any suggestions in this regard would be extremely helpful

**Comment by mtesauro** [comment]
Mailhog is only used to develop and test DefectDojo - it's not meant to be a SMTP gateway or otherwise send emails.  You'd need an SMTP service that will forward emails for DefectDojo to get email notifications to work.

There's an example of what to set in settings.py at https://github.com/DefectDojo/django-DefectDojo/blob/ae5ff9d0677be1db39daa2a6db4811bb2a3fc4da/dojo/settings/settings.dist.py#L886

Most of the configuration is on the SMTP server, not DefectDojo. You can use a command-line tool like telnet or mailx to test connectivity and credentials to send emails to help diagnose email sending issues.

**Reply by guptachandresh** [reply]
This configuration change was made & a valid one but still in logs it shows Connection Refused.

**Reply by mtesauro** [reply]
Seems like there's either network connectivity problems between the DefectDojo install and the SMTP service. Since the connection was refused, it shows that DefectDojo is reaching out to the SMTP service and that service won't accept the connection. Thus, I'd look at the SMTP service or the network between DefectDojo and it.

**Reply by guptachandresh** [reply]
I already tried that using a simple python script and was successful in sending out emails. So connectivity is not an issue as well

**Reply by guptachandresh** [reply]
Quick update:
I was able to use godojo to deploy it on a vm and was successful in getting the email notifications working.
The only challenge was even though the config file had the email configuration it did not get copied to the settings.dist.py file. If we update it there then only it starts working.

Still trying to figure out how I can get it working in Kubernetes.
